app/src/main/java/org/groundplatform/android/persistence/local/room/stores/RoomSurveyStore.kt

```diff
@@ -69,9 +69,12 @@ class RoomSurveyStore @Inject internal constructor() : LocalSurveyStore {
    */
   override suspend fun insertOrUpdateSurvey(survey: Survey) =
     localDatabase.withTransaction {
+      // Update survey.
       surveyDao.insertOrUpdate(survey.toLocalDataStoreObject())
-      jobDao.deleteBySurveyId(survey.id)
+      // Add or update jobs and tasks.
       insertOrUpdateJobs(survey.id, survey.jobs)
+      // Delete removed jobs.
+      jobDao.deleteNotIn(survey.jobs.map { it.id })
```

Comment:
IIUC, this will remove jobs of other surveys as well. Is that intentional?

Reply:
Yikes, good catch!